### ISSUE TITLE
Flatpak / libostree fail to build on musl

### ISSUE THREAD

**Issue by aux-j** [fosero/flatpak-overlay#28]
Hi there, 

Flatpak from the flatpak-overlay won't currently build on musl. A working patch (from alpine) is here:

```
--- a/config.h.in
+++ b/config.h.in
@@ -140,3 +140,14 @@
 
 /* Define to 1 if you need to in order for `stat' and other things to work. */
 #undef _POSIX_SOURCE
+
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
```
 
Also, for sys-fs/libostree:

```
--- a/config.h.in
+++ b/config.h.in
@@ -140,3 +140,14 @@
 
 /* Define to 1 if you need to in order for `stat' and other things to work. */
 #undef _POSIX_SOURCE
+
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
```

**Comment by fosero** [comment]
Is this reported upstream?

Although I can see your issue, I am not keen on picking up patches that are not going to get in upstream.

**Comment by aux-j** [comment]
Hi,

I haven’t reported it upstream so I’m not sure if someone else has - but I imagine it’s been patched in other musl distros (void, alpine etc) to get flatpak built. 

I understand completely, I’ll have a look and report it to upstream if it hasn’t already been done!--
Tutanota


16 Feb 2019, 13:05 by notifications@github.com:

**Comment by fosero** [comment]
@gitlab-j Did you report it upstream and if so is there a issue number to track?

**Comment by ghost** [comment]
Can confirm that this is still happening. The patches above still work.

**Comment by okias** [comment]
Upstream issue: https://github.com/flatpak/flatpak/issues/618

**Comment by fosero** [comment]
I'm not quite sure this is fixed in 2019.5, but 2020.3 should have it.
